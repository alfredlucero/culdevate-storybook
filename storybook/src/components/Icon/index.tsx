import * as React from 'react';

import { IconType } from '../../types/icons';

export interface IconProps {
  type: IconType
}

const Icon: React.SFC<IconProps> = ({
  type,
}) => {
  return (
    <i className={`fa fa-${type}`}></i>
  );
};

export default Icon;
