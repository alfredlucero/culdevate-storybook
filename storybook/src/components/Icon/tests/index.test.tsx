import * as React from 'react';
import { shallow } from 'enzyme';
import * as renderer from 'react-test-renderer';
import 'jest-styled-components';

import Icon from '../index';

describe('<Icon />', () => {
  it('should render icon type correctly', () => {
    const tree = renderer.create(<Icon type="github"/>).toJSON();
    expect(tree).toMatchSnapshot();
  });
});

