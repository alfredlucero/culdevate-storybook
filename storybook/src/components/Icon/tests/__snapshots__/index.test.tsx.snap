// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Icon /> should render icon type correctly 1`] = `
<i
  className="fa fa-github"
/>
`;
